### ISSUE TITLE
Interactability_Effect does not work with "Spawn_On_Dedicated_Server" flag

### ISSUE THREAD

**Issue by DrDallas1** [SmartlyDressedGames/Unturned-3.x-Community#3703]
Hello,

The Interactability_Effect reward on interactable object, by default always spawns effects as a client side effect only, does not replicate to clients, even if you have the "Spawn_On_Dedicated_Server" flag in the Effect dat. I am not sure if this is intentional, but I am pretty sure this is a bug.

(The effect itself works, only for the player that calls it)

![image](https://user-images.githubusercontent.com/66083394/218626662-d05208a4-6106-4569-8b19-998fe7e5d380.png)

![image](https://user-images.githubusercontent.com/66083394/218626842-e6e99a5a-b376-4c9e-bc5b-d5740476bd3f.png)

**Comment by SDGNelson** [comment]
I don't think this is a bug per se, but I see how it could be useful and expected to play the effect for all players.

**Comment by DrDallas1** [comment]
I think you should be able to specify. Likely with the "Spawn_On_Dedicated_Server" flag already present on effects. Having a lot of previously client side only effects spawn server side could be disasterous.

**Comment by prank0079** [comment]
I think you can achieve your goal faster by using NpcGlobalEventHook with EffectSpawner than by waiting for the update.

**Comment by SDGNelson** [comment]
For the next update I've added an `InteractableObjectQuestEventHook` component which can be used with the Quest, Note, and Dropper interactable objects. An effect spawner component could be connected to it to spawn an effect for all players. I decided to go this route because the spawner component has a lot more options than would make sense to add to the object asset.

**Comment by DrDallas1** [comment]
@SDGNelson 
That is spectacular! It opens up so many new possibilities for quests and server side quest implementation. Can this event hook also work with IOBS (binary states)?

**Comment by SDGNelson** [comment]
What do you have in mind for working with IOBS? (I would think the existing IOBS Event Hook already has the needed functionality.)
